spaceship/lib/spaceship/portal/merchant.rb

```diff
@@ -0,0 +1,90 @@
+module Spaceship
+  module Portal
+    # Represents a Merchant ID from the Developer Portal
+    class Merchant < PortalBase
+      # @return (String) The identifier of this merchant, provided by the Dev Portal
+      # @example
+      #   "LM1UX73BAC"
+      attr_accessor :merchant_id
+
+      # @return (String) The name you provided for this merchant
+      # @example
+      #   "Spaceship Production"
+      attr_accessor :name
+
+      # @return (String) the supported platform of this merchant
+      # @example
+      #   "ios"
+      attr_accessor :platform
+
+      # Prefix provided by the Dev Portal
+      # @example
+      #   "5A9972XTK2"
+      attr_accessor :prefix
+
+      # @return (String) The bundle_id (merchant identifier) of merchant id
+      # @example
+      #   "merchant.com.krausefx.app.production"
+      attr_accessor :bundle_id
+
+      # @return (String) Status of the merchant
+      # @example
+      #   "current"
+      attr_accessor :status
+
+      attr_mapping(
+        'omcId' => :merchant_id,
+        'name' => :name,
+        'prefix' => :prefix,
+        'identifier' => :bundle_id,
+        'status' => :status
+      )
+
+      class << self
+        # Create a new object based on a hash.
+        # This is used to create a new object based on the server response.
+        def factory(attrs)
```

Comment:
Please remove usage of `factory`. It has been copied into too many places (no judgement, I recognize it's natural to reference existing implementations).

The usage of `factory` should be limited to very specific use-cases described here: https://github.com/fastlane/fastlane/blob/3ffaaf2d0befc2df3132b7c266bd775aa13f49b7/spaceship/lib/spaceship/base.rb#L177